python/ray/experimental/tf_utils.py

```diff
@@ -195,3 +200,353 @@ def set_weights(self, new_weights):
                 for (name, value) in new_weights.items()
                 if name in self.placeholders
             })
+
+
+DEBUG = 0
+def logger_debug_info(txt):
+    if DEBUG:
+        logger.info(txt)
+
+banner_half = "".join(["#"] * 10)
+
+def tf_differentiable(num_return_vals):
```

Comment:
The important functions in this file need a lot of documentation to explain precisely what they're doing (so that someone else could deeply understand it and modify the code).